src/components/diagrams/diagram.js

```diff
@@ -0,0 +1,911 @@
+                    const breakerToSwitchDom = document.getElementById(
+                        locallySwitchedBreaker.id
+                    );
+                    if (
+                        breakerToSwitchDom.classList.value.includes(
+                            'sld-closed'
+                        )
+                    ) {
+                        breakerToSwitchDom.classList.replace(
+                            'sld-closed',
+                            'sld-open'
+                        );
+                    } else if (
+                        breakerToSwitchDom.classList.value.includes('sld-open')
+                    ) {
+                        breakerToSwitchDom.classList.replace(
+                            'sld-open',
+                            'sld-closed'
+                        );
+                    }
+                }
+
+                //if original sld size has not changed (sld structure has remained the same), we keep the same zoom
+                if (
+                    svgDraw.current &&
+                    hasDiagramSizeRemainedTheSame(
+                        svgDraw.current.getOriginalWidth(),
+                        svgDraw.current.getOriginalHeight(),
+                        sldViewer.getOriginalWidth(),
+                        sldViewer.getOriginalHeight()
+                    )
+                ) {
+                    sldViewer.setViewBox(svgDraw.current.getViewBox());
+                }
+
+                // on sld resizing, we need to refresh zoom to avoid exceeding max or min zoom
+                // this is due to a svg.panzoom.js package's behaviour
+                sldViewer.refreshZoom();
+
+                svgUrl.current = svg.svgUrl;
+                svgDraw.current = sldViewer;
+            } else {
+                console.error('diagramType manquant #2');
+                alert('check console');
+            }
+        }
+    }, [
+        network,
+        diagramId,
+        props.svgUrl,
+        svg,
+        currentNode,
+        onNextVoltageLevelClick,
+        onBreakerClick,
+        isComputationRunning,
+        isAnyNodeBuilding,
+        equipmentMenu,
+        showEquipmentMenu,
+        svgType,
+        theme,
+        ref,
+        svgFinalHeight,
+        svgFinalWidth,
+        disabled,
+        modificationInProgress,
+        loadingState,
+        locallySwitchedBreaker,
+        diagramType,
+    ]);
+
+    useLayoutEffect(() => {
+        if (
+            typeof svgFinalWidth != 'undefined' &&
+            typeof svgFinalHeight != 'undefined'
+        ) {
+            const divElt = svgRef.current;
+            if (divElt != null) {
+                const svgEl = divElt.getElementsByTagName('svg')[0];
+                if (svgEl != null) {
+                    svgEl.setAttribute('width', svgFinalWidth);
+                    svgEl.setAttribute(
+                        'height',
+                        computedHeight ? computedHeight : svgFinalHeight
```

Comment:
you can remplace it by `computedHeight ?? svgFinalHeight`

Reply:
done